File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {

        int n = nums.length;

        int leftProd = 1;
        int rightProd = 1;
        int ans = nums[0];
        for(int i=0;i<n;i++){
            leftProd = leftProd == 0 ? 1 : leftProd;
            rightProd = rightProd == 0 ? 1 : rightProd;

            leftProd = leftProd * nums[i];
            rightProd = rightProd * nums[n-1-i]; // last element - the curent ith element

            ans  = Math.max(ans, Math.max(leftProd, rightProd));
        }

        return ans;
        
    }
}

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProduct = function(nums) {

    let max = -Infinity;

    for(let i=0;i<nums.length;i++){
        let prod = 1;
        for(let j=i;j<nums.length;j++){
            prod *= nums[j];
            if(prod > max) max = prod;
        }
    }

    return max;
    
};